apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -332,8 +333,26 @@ const Ticket = new GQLListSchema('Ticket', {
                 await calculateReopenedCounter(context, existingItem, resolvedData)
             }
 
+            // When creating ticket or updating ticket address,
+            // if client is not passed in resolvedData,
+            // we find a registered user with a phone number that matches the contact's phone number
+            // and an address that matches the ticket address.
+            if (userType === STAFF) {
+                const contactId = get(resolvedData, 'contact', null)
+                const propertyId = get(resolvedData, 'property', null)
+                const unitName = get(resolvedData, 'unitName', null)
+                const client = get(resolvedData, 'client', null)
+
+                if (isNull(client) && (!isNull(contactId) || !isNull(propertyId) || !isNull(unitName))) {
+                    await setClientIfContactPhoneAndTicketAddressMatchesResidentFields(operation, resolvedData, existingItem)
+                }
+            }
+
             if (userType === RESIDENT && operation === 'create') {
-                await addClientInfoToResidentTicket(context, resolvedData)
+                await createContactIfNotExists(context, resolvedData)
```

Comment:
can we move contact creation to `afterChange` hook ?

Reply:
as for me it doesn't matter